### ISSUE TITLE
Intern reporters stopped emiting "testEnd" event

### ISSUE THREAD

**Issue by bartoszkaczorek** [theintern/intern#691]
## Overview

Intern reporters stopped emiting "testEnd" event. This worked OK in intern "3.3.2" (looks to be introduced with intern "3.4.0").

## Environment

intern: 3.4.1
OS: Windows 7 64 bit
browser: Firefox 47.0.1
nodejs: v4.4.4

## Steps to reproduce

1. Create file structure as shown in next chapter
2. Install intern 3.4.1:
`npm install`
3. Start selenium standalone server 2.53.0 on port 444
4. Run tests:
`npm test` 

Output from intern 3.4.1:
```
> intern-runner -config=test/config/intern.js
started: firefox on WINDOWS - unit tests - test/unit/exampleTests - When adding two numbers - Then result should contain sum of them
started: firefox on WINDOWS - unit tests - test/unit/exampleTests - When multiplying two numbers - Then result should contain product of them
```

Output from intern 3.3.2:
```

> intern-runner -config=test/config/intern.js
started: firefox on WINDOWS - unit tests - test/unit/exampleTests - When adding two numbers - Then result should contain sum of them
ended: firefox on WINDOWS - unit tests - test/unit/exampleTests - When adding two numbers - Then result should contain sum of them
started: firefox on WINDOWS - unit tests - test/unit/exampleTests - When multiplying two numbers - Then result should contain product of them
ended: firefox on WINDOWS - unit tests - test/unit/exampleTests - When multiplying two numbers - Then result should contain product of them
```

## File structure

*package.json*
```json
{
"name": "intern-reporter-test-end-bug",
"version" : "1.0.0",
"private": true,
"dependencies": {
"intern": "3.4.1"
},
"scripts" : {
"test": "node_modules\\.bin\\intern-runner -config=test/config/intern.js"
}
}
```

*test/config/intern.json*
```json
{
"proxyPort": 9000,
"proxyUrl": "http://localhost:9000/",
"capabilities": {
"selenium-version": "2.53.0"
},
"environments": [{
"browserName": "firefox",
"platform": ["WINDOWS"]
}],
"tunnelOptions": {
"hostname": "localhost",
"port": 4444
},
"suites": ["test/unit/**/*"],
"excludeInstrumentation": "(node_modules|test)[\\\\/]",
"reporters": ["test/reporter/customReporter"]
}
```

_test/unit/intern.js_
```js
define(['intern/dojo/text!./intern.json'], function (configJson) {
"use strict";

var config = JSON.parse(configJson);
if (typeof config.excludeInstrumentation === 'string') {
config.excludeInstrumentation = new RegExp(config.excludeInstrumentation);
}

return config;
});
```

_test/unit/exampleTests.js_
```js
define(['module',
'intern!bdd',
'intern/chai!expect'], function (module, bdd, expect) {
"use strict";

bdd.describe(module.id, function () {

bdd.describe('When adding two numbers', function () {
var result = null;

bdd.before(function () {
result = 2 + 3;
});

bdd.it('Then result should contain sum of them', function () {
expect(result).to.equal(5);
});
});

bdd.describe('When multiplying two numbers', function () {
var result = null;

bdd.before(function () {
result = 4 * 5;
});

bdd.it('Then result should contain product of them', function () {
expect(result).to.equal(20);
});
});

});
});

```

_test/reporter/customReporter.js_
```js
define(function (require) {
  return {
    '/test/start': function (test) {
    console.log('started: ' + test.id);
    },

  '/test/end': function (test) {
    console.log('ended: ' + test.id);
    }
  };
});
```

**Comment by jason0x43** [comment]
The `testEnd` event is being emitted by the reporter manager for tests run locally on the Intern host, but not for unit tests run in a remote browser. It's being lost somewhere in the WebDriver message path.

**Comment by jason0x43** [comment]
But as you say, this wasn't the case in 3.3.2, so something is broken in 3.4.

**Comment by jason0x43** [comment]
Commit b622530d441299a7bfc7e5b40ac77dd4493c609d appears to be the culprit.

**Comment by jason0x43** [comment]
Fixed in d1cac5826185254b8c7108e104e24c73f3ee64a7
